pkg/profileparser/profileparser.go

```diff
@@ -199,17 +199,20 @@ func createOrUpdate(cli runtimeclient.Client, kind string, key types.NamespacedN
 	return nil
 }
 
-func deleteObsoleteItems(cli runtimeclient.Client, kind string, namespace string, imageDigest string) error {
+func deleteObsoleteItems(cli runtimeclient.Client, kind string, pbName, namespace string, imageDigest string) error {
 	list := unstructured.UnstructuredList{}
 	list.SetGroupVersionKind(schema.GroupVersionKind{
 		Group:   cmpv1alpha1.SchemeGroupVersion.Group,
 		Version: cmpv1alpha1.SchemeGroupVersion.Version,
 		Kind:    kind + "List",
 	})
 	inNs := runtimeclient.InNamespace(namespace)
+	withPbOwnerLabel := runtimeclient.MatchingLabels{
+		cmpv1alpha1.ProfileBundleOwnerLabel: pbName,
+	}
 
-	log.Info("Checking for unused object", "kind", kind, "namespace", namespace)
-	if err := cli.List(context.TODO(), &list, inNs); err != nil {
+	log.Info("Checking for unused object", "kind", kind, "owner", pbName, "namespace", namespace)
+	if err := cli.List(context.TODO(), &list, inNs, withPbOwnerLabel); err != nil {
```

Comment:
Why not refactor this to use `DeleteAllOf` as done here https://github.com/openshift/compliance-operator/pull/396 ?

Reply:
Note that you'd need to add an extra permission via RBAC to do this.

Reply:
Ah, OK, I remembered why I didn't use DeleteAllOf..the hashes are stored as annotations because a typical sha256 hash is longer than what labels would allow. And typically the fields you use as DeleteOptions or ListOptions must be indexed, annotations are not. It appears that it's possible to index annotations and then use MatchingFields (e.g. https://github.com/kubermatic/utils/blob/39ab8a8d6830548e12f069ad75216c2849065907/pkg/multiowner/multi_owner.go) but I'm not sure if it's a good pattern.

But that brings a more important quesition: do you think it's OK to use annotations and loop? My thought when I wrote the code was that updating a profile is a rare operation and not really performance sensitive. But if you don't like that we can either index the annotations or somehow put the sha hash into the labels (e.g. quay or git use shortened hash versions, but I wasn't able to find the details. I haven't looked too deep though tbh)